internal/controller/postgrescluster/postgres_test.go

```diff
@@ -414,8 +414,14 @@ volumeMode: Filesystem
 					assert.Assert(t, returned == nil)
 
 					key, fetched := client.ObjectKeyFromObject(pvc), &corev1.PersistentVolumeClaim{}
-					assert.NilError(t, tClient.Get(ctx, key, fetched))
-					assert.Assert(t, fetched.DeletionTimestamp != nil, "expected deleted")
+					err = tClient.Get(ctx, key, fetched)
+					assert.NilError(t, client.IgnoreNotFound(err))
+					// If we have a NotFound error the item has been deleted
+					if !apierrors.IsNotFound(err) {
+						// If we don't have a NotFound error, the item still exists
+						// check that the item is marked for deletion
+						assert.Assert(t, fetched.DeletionTimestamp != nil, "expected deleted")
+					}
```

Comment:
🤔 I thought we patched this already, but I guess not.

The checks for `IgnoreNotFound` and `IsNotFound` are a little confusing. Is the following any better? It is more clear to me that we look in `fetched` when it is populated.

```suggestion
					if err := tClient.Get(ctx, key, fetched); err == nil {
						assert.Assert(t, fetched.DeletionTimestamp != nil, "expected deleted")
					} else {
						assert.Assert(t, apierrors.IsNotFound(err), "expected NotFound, got %v", err)
					}
```

Reply:
I think that is more clear :+1: